sdk/identity/azure-identity/conftest.py

```diff
@@ -6,18 +6,31 @@
 import sys
 
 import pytest
-from azure.identity._constants import EnvironmentVariables
+from azure.identity._constants import AZURE_CLI_CLIENT_ID, EnvironmentVariables
 
-# IMDS tests must be run explicitly
-collect_ignore_glob = ["*imds*"]  # pylint:disable=invalid-name
-
-# Ignore collection of async tests on unsupported platforms
+# Ignore async tests on unsupported platforms
 if sys.version_info < (3, 5):
-    collect_ignore_glob.append("*_async.py")
+    collect_ignore_glob = ["*_async.py"]
+
+run_manual_tests = "manual" in sys.argv
```

Comment:
"paranoia-for-future-me-being-dumb" comment incoming:  Your commit message gives good detail on usage of these flags, but as an outsider coming in, I'm not sure if it's at all discoverable.  Is there a readme/other source of knowledge for identity tests?  (in a perfect world I'd want to plumb this into proper args and give it details in -h, but I would guess that's hard because identity is just one possible configuration within our generic pytest setup?)

Reply:
This particular line is an ugly hack to determine whether pytest was launched with '-m manual', so we can use the 'manual' marker to opt in to manual tests. To your broader point, all knowledge outside the code is tribal. I have a tracking issue for documenting the test matrix (#8368) but there's higher priority test work likely to obsolete any document written today, e.g. setting up recording infrastructure (#5980).